step/codesign.go

```diff
@@ -106,10 +108,17 @@ func createCodesignManager(managerOpts CodesignManagerOpts, xcodeMajorVersion in
 	}
 
 	client := retry.NewHTTPClient().StandardClient()
+
 	var testDevices []devportalservice.TestDevice
-	if serviceConnection != nil {
+	if managerOpts.TestDeviceListPath != "" {
+		testDevices, err = devportalservice.ParseTestDevicesFromFile(managerOpts.TestDeviceListPath, time.Now())
+		if err != nil {
+			logger.Warnf("Failed to process device list (%s): %s", managerOpts.TestDeviceListPath, err)
```

Comment:
Shouldn't it be an actual error, this open has precedence over the registered test devices on Bitrise and this error can only occur if the user intentionally provided the input.

Reply:
Good idea. I was thinking of that if we could not register the device then the build could succeed and generate artefacts but your reasoning is good. The only way we can encounter an error is when the user set the value of this input on purpose. So they expect that the device registration will happen and if it does not then it is reasonable to throw an error.

I like it and will change the code.